pkg/apis/elasticsearch/v1beta1/elasticsearch_types.go

```diff
@@ -106,54 +107,59 @@ func (n NodeSet) GetESContainerTemplate() *corev1.Container {
 type ChangeBudget struct {
-	// TODO: MaxUnavailable and MaxSurge would be great to have as intstrs, but due to
-	// https://github.com/kubernetes-sigs/kubebuilder/issues/442 this is not currently an option.
-
-	// MaxUnavailable is the maximum number of pods that can be unavailable during the update.
-	// Value can be an absolute number (ex: 5) or a percentage of total pods at the start of update (ex: 10%).
-	// Absolute number is calculated from percentage by rounding down.
-	// This can not be 0 if MaxSurge is 0 if you want automatic rolling changes to be applied.
-	// By default, a fixed value of 0 is used.
-	// Example: when this is set to 30%, the group can be scaled down by 30%
-	// immediately when the rolling update starts. Once new pods are ready, the group
-	// can be scaled down further, followed by scaling up the group, ensuring
-	// that at least 70% of the target number of pods are available at all times
-	// during the update.
-	MaxUnavailable int `json:"maxUnavailable"`
-
-	// MaxSurge is the maximum number of pods that can be scheduled above the original number of
-	// pods.
+	// MaxUnavailable is the maximum number of pods that can be unavailable (not ready) during the update due to the
+	// actions controlled by the operator.
 	// By default, a fixed value of 1 is used.
-	// Value can be an absolute number (ex: 5) or a percentage of total pods at
-	// the start of the update (ex: 10%). This can not be 0 if MaxUnavailable is 0 if you want automatic rolling
-	// updates to be applied.
-	// Absolute number is calculated from percentage by rounding up.
-	// Example: when this is set to 30%, the new group can be scaled up by 30%
-	// immediately when the rolling update starts. Once old pods have been killed,
-	// new group can be scaled up further, ensuring that total number of pods running
-	// at any time during the update is at most 130% of the target number of pods.
-	MaxSurge int `json:"maxSurge"`
+	MaxUnavailable *int32 `json:"maxUnavailable,omitempty"`
+
+	// MaxSurge is the maximum number of pods that can be scheduled above the original number of pods. MaxSurge
+	// is only taken into the account when scaling up.
+	// By default, it's unbounded.
+	MaxSurge *int32 `json:"maxSurge,omitempty"`
 }
 
 // DefaultChangeBudget is used when no change budget is provided. It might not be the most effective, but should work in
-// every case
+// most cases.
 var DefaultChangeBudget = ChangeBudget{
-	MaxSurge:       1,
-	MaxUnavailable: 0,
+	MaxSurge:       nil,
+	MaxUnavailable: common.Int32(1),
+}
+
+func (cb ChangeBudget) GetMaxSurgeOrDefault() *int32 {
+	// use default if not specified
+	maxSurge := DefaultChangeBudget.MaxSurge
+	if cb.MaxSurge != nil {
+		maxSurge = cb.MaxSurge
+	}
+
+	// nil or negative in the spec denotes unlimited surge
+	// in the code unlimited surge is denoted by nil
+	if maxSurge == nil || *maxSurge < 0 {
+		maxSurge = nil
+	}
+
+	return maxSurge
```

Comment:
Slightly simplified version (also applies to `GetMaxUnavailableOrDefault `):
```go
if cb.maxSurge != nil && cb.MaxSurge >=0 {
    return cb.maxSurge
}
return DefaultChangeBudget.MaxSurge
```

Reply:
Yes, but it will not work as expected if we change default to something else. Eg. user specifies -1 to get unbounded, but our default is 10.

Reply:
Got it :)